proposals/top-level-statements.md

```diff
@@ -81,15 +81,15 @@ static class Program
 }
 ```
 
-Note that the names "Program" and "Main" are used only for illustrations purposes, actual names used by
-compiler are implementation dependent and neither the type, nor the method can be referenced by name from
+Note that the names "Program" and "Main" are used only for illustration purposes, actual names used by the
+compiler are implementation-dependent and neither the type, nor the method can be referenced by name from
```

Comment:
Please remove the hyphen. `implementation` and `dependent` are not separate modifiers on `and`.

Reply:
I assume the writer intended to ascribe a single property, that of being dependent on the implementation, to `actual names used by the compiler`. Thus, I used the adjective `implementation-dependent`, **with hyphenation**, as it ascribes a single property to `actual names used by the compiler`.

Reply:
We do not use a hyphen for this. Please remove.

Reply:
Implementation-dependent is grammatically correct. I don't know if we have a style guide which says otherwise.